### PR TITLE
Cleanup node cloning and skip sandbox

### PR DESCRIPTION
**Pull request by IDisposable**
Root node is detected when `cloneNode` is passed `null` for  `parentComputedStyles`
Hoisted `sandbox` up higher
Don't ever clone the `sandbox` node
Remove spurious `util.delay(0)` in rendering chain
Removed extra `cloneChildrenInOrder` method and in-lined the body into `cloneChildren`
Made the references to the `original` element and its children and styles more apparent.

### CODE REVIEW

**Comment by IDisposable** [comment]
(sorry, fixed the base to be the 3.0 lineage.

**Comment by IDisposable** [comment]
I will likely rebase this once I get the v2 - dynasty branch stuff correct and make `master` the current branch for v3

File: src/dom-to-image-more.js [hunk 1]
```diff
@@ -206,7 +205,6 @@
         options = options || {};
         return toSvg(domNode, options)
             .then(util.makeImage)
-            .then(util.delay(0))
```
Comment (IDisposable) [inline src/dom-to-image-more.js:209]: This was put in a VERY long time ago to fix some yielding issues. It needs to go away.

File: src/dom-to-image-more.js [hunk 2]
```diff
@@ -1018,7 +1007,8 @@
         const defaultElement = document.createElement(tagName);
         sandbox.contentWindow.document.body.appendChild(defaultElement);
         // Ensure that there is some content, so that properties like margin are applied.
-        defaultElement.textContent = '.';
+        // we use zero-width space to handle FireFox adding a pixel
+        defaultElement.textContent = '\0x200B';
```
Comment (IDisposable) [inline src/dom-to-image-more.js:1011]: Not sure if this is better or worse, but as long as we're not trying to inject `SCRIPT` tags, it shouldn't matter anymore :)

Comment (meche-gh) [reply src/dom-to-image-more.js:1011]: To get the string working in my console:
```suggestion
        defaultElement.textContent = '\u200b';
```

Generally I think the ZWSP is better:
- we're now thinking about "empty" elements for `getDefaultStyle`
- the hack now narrowly targets the bug
- plus for free, we avoid a value collision when an element's CSS has the precise width of a `.` but is empty

**Comment by IDisposable** [comment]
Merged by hand.

File: src/dom-to-image-more.js (hunk 2 above)
Comment (IDisposable) [reply src/dom-to-image-more.js]: Applied this in another branch, thanks!

**Pull request closed without merging**